starter-repos/shared-ci-cd/appctl/validate.yaml

```diff
@@ -0,0 +1,35 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+validate-manifests:
+  stage: validate-manifests
+  image: gcr.io/cloud-solutions-images/appctl
+  script:
+  - |
+    cat > /root/.appctlconfig <<EOF
+    personal_access_tokens:
+    - host: ${CI_SERVER_HOST}
+      token: ${GIT_ACCESS_TOKEN}
+    EOF
+  - cat > /root/.ssh/id_rsa <<EOF
+    ${MANIFEST_WRITER_KEY}
+    EOF
+  - chmod 400 /root/.ssh/id_rsa
+  - echo "StrictHostKeyChecking no " > /root/.ssh/config
+  - git config --global user.email ${CI_PROJECT_NAME}-ci@${CI_SERVER_HOST}
+  - git config --global user.name ${CI_PROJECT_NAMESPACE}/${CI_PROJECT_NAME}
+  - |
+    appctl init ${APP_NAME} --app-config-repo ${APP_REPO_URL} \
+    --deployment-repo ${DEPLOYMENT_REPO_URL}
+  - appctl validate staging-central --from-revision=master
```

Comment:
Since this stage is supposed to run on all the branches, should `--from-revision` be set to something else other than `master`?

Reply:
Thanks for catching this. Set to the predefined `${CI_COMMIT_BRANCH}`